app/server/lib/FlexServer.ts

```diff
@@ -1883,6 +1883,19 @@ export class FlexServer implements GristServer {
     const probes = new BootProbes(this.app, this, '/api', adminMiddleware);
     probes.addEndpoints();
 
+    this.app.post('/api/admin/restart', requireInstallAdmin, expressWrap(async (req, resp) => {
+      const newEnv = req.body.newEnv;
+      resp.on('finish', () => {
+        // If we have IPC with parent process (e.g. when running under
+        // Docker) tell the parent that we have a new environment so it
+        // can restart us.
+        if (process.send) {
+          process.send({ newEnv });
+        }
+      });
+      return resp.status(200).send();
```

Comment:
Is it worth sending an error status code if we've got no process.send available (i.e - process isn't running under the runner script)

Reply:
Is there a risk we send sensitive environment variables over Redis in that situation?

Reply:
Having talked to Jordi a bit, I'm not very convinced by the idea of implementing Grist configuration changes from the UI directly via changing environment variables. Specifically, there's no concrete proposal yet for how those changes will be persisted (@jordigh mentioned @SleepyLeslie's config PR for Grist Desktop as interesting). If there were, that might help answer the question of how configuration is kept consistent across servers/workers.

Pub/sub in redis seems an obvious choice for triggering a global restart. It may or may not be a reasonable way to propagate configuration, I haven't thought about that. Redis is already used for security sensitive purposes related to authentication and authorization.

Until there is a persistence story: what state do we actually need to persist? If I'm reading the code right, if a user provides an activation key, there's a path to `activations.key` getting set in the db:
https://github.com/gristlabs/grist-ee/blob/e86b290a4c2c3cc3aea201ae4787ecb8dab4e3c0/ext/app/server/lib/ActivationReader.ts#L87-L90

What I'm poking at here is if we need env var persistence this month.

Reply:
Okay, did these changes.

Reply:
I've stubbed out config changes, which I think we should incorporate with [this PR](https://github.com/gristlabs/grist-desktop/pull/36) once it lands.